pkg/tests/ossm/setup.go

```diff
@@ -51,21 +51,36 @@ func installNightlyOperators() {
 	time.Sleep(time.Duration(30) * time.Second)
 }
 
-// Initialize a default SMCP and SMMR
-func SetupNamespacesAndControlPlane() {
+func BasicSetup() {
 	log.Log.Info("Setting up namespaces and OSSM control plane")
 	createNamespaces()
-
 	if env.Getenv("NIGHTLY", "false") == "true" {
 		installNightlyOperators()
 	}
-
 	util.ShellMuteOutputError(`oc new-project %s`, meshNamespace)
-	util.KubeApplyContents(meshNamespace, util.RunTemplate(smcpV23_template, smcp))
+}
+
+// Initialize a default SMCP and SMMR
+func SetupNamespacesAndControlPlane() {
+	BasicSetup()
+	versionTemplates := GetSMCPTemplates()
+	smcpVersion := env.GetDefaultSMCPVersion()
+	template, ok := versionTemplates[smcpVersion]
+	if !ok {
+		log.Log.Errorf("Unsupported SMCP version: %s", smcpVersion)
+		return
+	}
+	util.KubeApplyContents(meshNamespace, util.RunTemplate(template, smcp))
 	util.KubeApplyContents(meshNamespace, smmr)
 	util.Shell(`oc -n %s wait --for condition=Ready smcp/%s --timeout 180s`, meshNamespace, smcp.Name)
 	util.Shell(`oc -n %s wait --for condition=Ready smmr/default --timeout 180s`, meshNamespace)
 	if ipv6 == "true" {
 		log.Log.Info("Running the test with IPv6 configuration")
 	}
 }
+
+// Initialize a default SMCP and SMMR
+func SetupOnlyNamespaces() {
+	BasicSetup()
+	// TODO: add more setup steps for test who do not need SMCP. If you need to add more steps, please add them here. If not we can remove this function.
+}
```

Comment:
Why not remove this function and just call BasicSetup directly? If we later determine that we need more, we can always refactor.